src/mlpack/core/optimizers/CMakeLists.txt

```diff
@@ -1,41 +1,12 @@
 set(DIRS
-  ada_delta
-  ada_grad
-  adam
-  aug_lagrangian
-  cmaes
-  bigbatch_sgd
-  cne
-  fw
   gradient_descent
   grid_search
```

Comment:
These are both a part of ensmallen also; we could probably remove them from here too.  I think there was a slight change to the APIs of each, but I can adapt that code if you like.

Reply:
Right, both optimizer make use of mlpack's ``DatasetMapper`` class, which isn't part of ensmallen; I guess it would be nice to have some option to extend the optimizer class in the same way it is possible for armadillo, but I guess if we can adapt the existing code to use the class, there is no need.

Reply:
https://gist.github.com/rcurtin/bbb9aa2ff5648626d8bd373ec3227e5f should fix it.  It incorporates a patch that I just opened for ensmallen that hasn't been merged yet.